cli_047_ndc_ratification/contents/src/__init__.py

```diff
@@ -7,99 +7,258 @@
+    body = {
+        "dataLastUpdated": date.isoformat() # date should be a string in the format 'YYYY-MM-DDTHH:MM:SS'
+    }
+    # send the request
+    try:
+        r = requests.patch(url = apiUrl, json = body, headers = headers)
+        logging.info('[lastUpdated]: SUCCESS, '+ date.isoformat() +' status code '+str(r.status_code))
+        return 0
+    except Exception as e:
+        logging.error('[lastUpdated]: '+str(e))
+
+'''
+FUNCTIONS FOR CARTO DATASETS
+
+The functions below must go in every near real-time script for a Carto dataset.
+Their format should not need to be changed.
+'''
+
+def checkCreateTable(table, schema, id_field, time_field=''):
+    '''
+    Create the table if it does not exist, and pull list of IDs already in the table if it does
+    INPUT   table: Carto table to check or create (string)
+            schema: dictionary of column names and types, used if we are creating the table for the first time (dictionary)
+            id_field: name of column that we want to use as a unique ID for this table; this will be used to compare the
+                    source data to the our table each time we run the script so that we only have to pull data we
+                    haven't previously uploaded (string)
+            time_field:  optional, name of column that will store datetime information (string)
+    RETURN  list of existing IDs in the table, pulled from the id_field column (list of strings)
+    '''
+    # check it the table already exists in Carto
+    if cartosql.tableExists(table, user=CARTO_USER, key=CARTO_KEY):
+        # if the table does exist, get a list of all the values in the id_field column
+        logging.info('Fetching existing IDs')
+        r = cartosql.getFields(id_field, table, f='csv', post=True, user=CARTO_USER, key=CARTO_KEY)
+        # turn the response into a list of strings, removing the first and last entries (header and an empty space at end)
+        return r.text.split('\r\n')[1:-1]
+    else:
+        # if the table does not exist, create it with columns based on the schema input
+        logging.info('Table {} does not exist, creating'.format(table))
+        cartosql.createTable(table, schema, user=CARTO_USER, key=CARTO_KEY)
+        # if a unique ID field is specified, set it as a unique index in the Carto table; when you upload data, Carto
+        # will ensure no two rows have the same entry in this column and return an error if you try to upload a row with
+        # a duplicate unique ID
+        if id_field:
+            cartosql.createIndex(table, id_field, unique=True, user=CARTO_USER, key=CARTO_KEY)
+        # if a time_field is specified, set it as an index in the Carto table; this is not a unique index
+        if time_field:
+            cartosql.createIndex(table, time_field, user=CARTO_USER, key=CARTO_KEY)
+        # return an empty list because there are no IDs in the new table yet
+        return []
+
+def delete_local():
```

Comment:
This isn't really a Carto-specific function--it could be used for any dataset, so if we include this, we would put it in the section above. But actually, we don't really need it...remind me to talk to you about this function when we check in!
```suggestion
```